tutorial/opt/aero/plot_optHist.py

```diff
@@ -3,49 +3,92 @@
 import matplotlib.pyplot as plt
 from pyoptsparse import History
 
-plt.rcParams["text.usetex"] = True  # Comment out if latex installation is not present
-plt.rcParams["font.family"] = "serif"
-plt.rcParams["font.size"] = 20
-plt.rcParams["xtick.labelsize"] = 16
-plt.rcParams["ytick.labelsize"] = 16
-plt.rcParams["lines.linewidth"] = 3.0
 
-parser = argparse.ArgumentParser()
-parser.add_argument("--histFile", type=str, default="opt.hst")
-parser.add_argument("--outputDir", type=str, default="./")
-args = parser.parse_args()
+def main():
 
-optHist = History(args.histFile)
-histValues = optHist.getValues()
+    plt.rcParams["text.usetex"] = True  # Comment out if latex installation is not present
+    plt.rcParams["font.family"] = "serif"
+    plt.rcParams["font.size"] = 20
+    plt.rcParams["xtick.labelsize"] = 16
+    plt.rcParams["ytick.labelsize"] = 16
+    plt.rcParams["lines.linewidth"] = 3.0
 
-fig, axes = plt.subplots(nrows=5, sharex=True, constrained_layout=True, figsize=(14, 10))
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--histFile", type=str, default="opt.hst")
+    parser.add_argument("--opt", type=str, default="SLSQP", choices=["IPOPT", "SLSQP", "SNOPT"])
```

Comment:
The user shouldn't need to specify the optimizer. It is stored in the metadata field of the history file, which you can get with `hist.getMetadata()`, which is a dict containing a field called `optimizer`.

Reply:
Agree with this

Reply:
I have removed the args.opt from args parser and instead incorporated hist.getMetadata()['optimizer'] to check for the optimizer